assets/js/onboard.js

```diff
@@ -75,13 +75,71 @@ const progressPlannerOnboardCall = ( data ) => {
 };
 
 if ( document.getElementById( 'prpl-onboarding-form' ) ) {
+	document
+		.querySelectorAll( 'input[name="with-email"]' )
+		.forEach( ( input ) => {
+			input.addEventListener( 'change', function () {
+				if ( 'no' === this.value ) {
+					document
+						.getElementById( 'prpl-onboarding-form' )
+						.querySelectorAll( 'input' )
+						.forEach( ( inputField ) => {
+							inputField.required = false;
+						} );
+					document.querySelector(
+						'#prpl-onboarding-form .prpl-form-fields'
+					).style.display = 'none';
+					document.querySelector(
+						'#prpl-onboarding-form .prpl-button-primary'
+					).style.display = 'none';
+					document.querySelector(
+						'#prpl-onboarding-form .prpl-button-secondary--no-email'
+					).style.display = 'block';
```

Comment:
This doesn't look very DRY, especially as we do the same below. Can we dry it up a bit?